resources/sdk/pureclouddotnet/templates/Logger.mustache

```diff
@@ -0,0 +1,312 @@
+            {
+                LogFormat logFormatValue = (LogFormat) Enum.Parse(typeof(LogFormat), logFormat, true);
+                if (Enum.IsDefined(typeof(LogFormat), logFormatValue) | logFormatValue.ToString().Contains(","))
+                    return logFormatValue;
+                else
+                    return null;
+            }
+            catch (ArgumentException)
+            {
+                return null;
+            }
+        }
+
+        public void Trace(string method,
+                          string url,
+                          object requestBody,
+                          int statusCode,
+                          Dictionary<String, String> requestHeaders,
+                          Dictionary<String, String> responseHeaders)
+        {
+            LogStatement logStatement = new LogStatement(
+                DateTime.UtcNow,
+                "trace",
+                method,
+                url,
+                requestHeaders,
+                responseHeaders,
+                statusCode,
+                requestBodyToString(requestBody)
+            );
+
+            log(LogLevel.LTrace, logStatement);
+        }
+
+        public void Error(string method,
+                          string url,
+                          object requestBody,
+                          string responseBody,
+                          int statusCode,
+                          Dictionary<String, String> requestHeaders,
+                          Dictionary<String, String> responseHeaders)
+        {
+            LogStatement logStatement = new LogStatement(
+                DateTime.UtcNow,
+                "error",
+                method,
+                url,
+                requestHeaders,
+                responseHeaders,
+                statusCode,
+                requestBodyToString(requestBody),
+                responseBody
+            );
+
+            log(LogLevel.LError, logStatement);
+        }
+
+        private void log(LogLevel logLevel, LogStatement logStatement)
+        {
+            if (logLevel >= LogLevel)
+            {
+                string logString = logStatement.AsString(LogFormat, LogRequestBody, LogResponseBody);
+                if (LogToConsole)
+                    Console.WriteLine(logString);
+
+                if (fs != null)
+                {
+                    try
+                    {
+                        byte[] info = new UTF8Encoding(true).GetBytes(String.Format("{0}\n", logString));
+                        fs.Write(info, 0, info.Length);
+                        fs.Flush();
+                    }
+                    catch (Exception)
+                    {
+                        // no-op
+                    }
+                }
+            }
+        }
+
+        private static string requestBodyToString(object requestBody)
+        {
+            if (requestBody != null)
+            {
+            //     if (requestBody.GetType() == typeof(String))
```

Comment:
Why is this code commented out?  Remove uncommented code unless you are planning to use it.